ros2service/ros2service/verb/echo.py

```
@@ -0,0 +1,236 @@
+
+        if self.srv_module is None:
+            raise RuntimeError(
+                'Could not load the type for the passed service')
+
+        event_topic_name = args.service_name + "/_service_event"
+
+        with NodeStrategy(args) as node:
+            self.subscribe_and_spin(
+                node,
+                event_topic_name,
+                self.event_msg_type
+            )
+
+    def subscribe_and_spin(
+            self,
+            node: Node,
+            event_topic_name: str,
+            event_msg_type: MsgType,
+    ) -> Optional[str]:
+        """Initialize a node with a single subscription and spin."""
+
+        node.create_subscription(
+            event_msg_type,
+            event_topic_name,
+            self._subscriber_callback,
+            self.qos_profile)
+
+        rclpy.spin(node)
+
+    def _subscriber_callback(self, msg, info):
+        deserialize_msg_type = None
+        self.event_enum = msg.info.event_type
+
+        if self.client and self.server:
+            pass
+        elif self.client:
+            if self.event_enum is ServiceEventType.REQUEST_RECEIVED or \
+                    self.event_enum is ServiceEventType.RESPONSE_SENT:
+                return
+        elif self.server:
+            if self.event_enum is ServiceEventType.REQUEST_SENT or \
+                    self.event_enum is ServiceEventType.RESPONSE_RECEIVED:
+                return
+
+        if self.event_enum is ServiceEventType.REQUEST_RECEIVED or \
+                self.event_enum is ServiceEventType.REQUEST_SENT:
+            deserialize_msg_type = self.srv_module.Request
+        elif self.event_enum is ServiceEventType.RESPONSE_RECEIVED or \
+                self.event_enum is ServiceEventType.RESPONSE_SENT:
+            deserialize_msg_type = self.srv_module.Response
+        else:  # TODO remove this once event enum is correct
+            print("Returning invalid service event type")
+            return
```

Should this case be removed? If we get to this case it sounds like it implies an error in the service introspection implementation. I think we should probably raise an error here, like
```py
raise ValueError(f'received unexpected service event type {event_enum}')
```